File: system-test/sec-tests/python/sec_21_file_upload.py
# SEC-21: Unrestricted File Upload
# Test: TC-CR-028
from flask import Flask, request

app = Flask(__name__)

@app.route('/upload', methods=['POST'])
def upload():
    file = request.files['file']
    file.save('/uploads/' + file.filename)
    return 'uploaded'

File: system-test/sec-tests/typescript/sec_21_file_upload.ts
// SEC-21: Unrestricted File Upload
// Test: TC-CR-028
import express from 'express';
import multer from 'multer';
import fs from 'fs';

const app = express();
const upload = multer({ dest: '/uploads/' });

app.post('/upload', upload.single('file'), (req: any, res) => {
  const file = req.file;
  fs.renameSync(file.path, '/uploads/' + file.originalname);  // No type/size check
  res.send('uploaded');
});
